Fix preserving timestamp when the move command is executing

KostiantynPopovych opened the pull request:
This PR resolves issue: #992.

These test files are pretty similar, I just took the `copy` method test files as a blueprint, and it seems we can make some refactoring here and there. I'm open to any suggestions!

RyanZim commented:
https://github.com/jprichardson/node-fs-extra/issues/992#issuecomment-1414186359

KostiantynPopovych commented:
Hey @RyanZim, I updated the implementation to match the logic that was discussed in thread #992. I refactored a bit methods move & moveSync to match the same style.

RyanZim commented:
It's not clear to me why the large refactor is needed; theoretically you only need to set `preserveTimestamps` in `moveAcrossDevice`, similarly to how `errorOnExist` is currently set. What am I missing?

KostiantynPopovych commented:
@RyanZim Reverted original code, that refactoring might be overengineering for now 🤔

RyanZim commented on lib/move/__tests__/move-preserve-timestamp.test.js:
```diff
@@ -0,0 +1,97 @@
+
+/* global beforeEach, afterEach, describe, it */
+
+if (process.arch === 'ia32') console.warn('32 bit arch; skipping copy timestamp tests')
```
```suggestion
if (process.arch === 'ia32') console.warn('32 bit arch; skipping move timestamp tests')
```

RyanZim commented on lib/move/__tests__/move-preserve-timestamp.test.js:
```diff
@@ -0,0 +1,97 @@
+
+  const _it = __skipTests ? it.skip : it
+
+  afterEach(done => {
+    fse.removeSync(TEST_DIR)
+    fse.removeSync(SRC)
+    done()
```
Don't need `done` for synchronous hooks
```suggestion
  afterEach(() => {
    fse.removeSync(TEST_DIR)
    fse.removeSync(SRC)
```

RyanZim commented on lib/move/__tests__/move-sync-preserve-timestamp.test.js:
```diff
@@ -0,0 +1,95 @@
+
+/* global beforeEach, afterEach, describe, it */
+
+if (process.arch === 'ia32') console.warn('32 bit arch; skipping copy timestamp tests')
```
```suggestion
if (process.arch === 'ia32') console.warn('32 bit arch; skipping move timestamp tests')
```

RyanZim commented on lib/move/__tests__/move-sync-preserve-timestamp.test.js:
```diff
@@ -0,0 +1,95 @@
+
+  const _it = __skipTests ? it.skip : it
+
+  afterEach(done => {
+    fse.removeSync(TEST_DIR)
+    fse.removeSync(SRC)
+    done()
```
```suggestion
  afterEach(() => {
    fse.removeSync(TEST_DIR)
    fse.removeSync(SRC)
```

RyanZim reviewed (commented):
A few nits in the tests; otherwise looks good.

KostiantynPopovych commented:
Thank you for reviewing it!
